### PR TITLE
checker: add a flag require SELECT ON *.*

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #7115

### What is changed and how it works?

service can use `RunCheckOnConfigs` with `dumpWholeInstance = true` to require SELECT ON *.* privilege

#### Tests 

 - Unit test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: pkg/container/sortmap/sort.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+package sortmap
+
+import (
+	"sort"
+
+	"golang.org/x/exp/constraints"
+)
+
+// KV represents the KV pairs in the input map of Sort.
+type KV[K constraints.Ordered, V any] struct {
```
Comment (D3Hunter) [inline pkg/container/sortmap/sort.go:23]: maybe `Pair`

File: dm/pkg/checker/privilege.go [hunk 2]
```diff
@@ -238,135 +271,132 @@ func VerifyPrivileges(grants []string, lackPriv map[mysql.PrivilegeType]map[stri
...
 					continue
 				}
-				if _, ok := lackPriv[privElem.Priv]; !ok {
+				privs, ok := lackPrivs[privElem.Priv]
+				if !ok {
+					continue
+				}
+				if privs.needGlobal {
+					continue
+				}
```
Comment (D3Hunter) [inline dm/pkg/checker/privilege.go:363]:
```suggestion
				if !ok  || privs.needGlobal {
					continue
				}
```

others too

**Review by D3Hunter — APPROVED** [review]

File: dm/checker/check_test.go [hunk 3]
```diff
@@ -67,7 +67,7 @@ func TestIgnoreAllCheckingItems(t *testing.T) {
 	require.NoError(t, err)
 	require.Len(t, msg, 0)
 
-	result, err := RunCheckOnConfigs(context.Background(), nil)
+	result, err := RunCheckOnConfigs(context.Background(), nil, false)
```
Comment (buchuitoudegou) [inline dm/checker/check_test.go:70]: Do we need a test where `dumpWholeInstance ` is true?

File: dm/checker/checker.go [hunk 4]
```diff
@@ -217,7 +218,13 @@ func (c *Checker) Init(ctx context.Context) (err error) {
 				if err != nil {
 					return err
 				}
-				c.checkList = append(c.checkList, checker.NewSourceDumpPrivilegeChecker(instance.sourceDB.DB, instance.sourceDBinfo, checkTables, exportCfg.Consistency))
+				c.checkList = append(c.checkList, checker.NewSourceDumpPrivilegeChecker(
+					instance.sourceDB.DB,
+					instance.sourceDBinfo,
+					checkTables,
+					exportCfg.Consistency,
+					false,
```
Comment (buchuitoudegou) [inline dm/checker/checker.go:226]:
```suggestion
				c.dumpWholeInstance,
```
I don't understand why it is not `c.dumpWholeInstance `

**Review by buchuitoudegou — COMMENTED** [review]
Will review the rest later

File: dm/checker/check_test.go (hunk 3 above)
Comment (lance6716) [reply dm/checker/check_test.go:70]: https://github.com/pingcap/tiflow/pull/7331/files#diff-f908b4efd00a69aa510a8c86679d44f92bb01f08ec21c0a653f4abe0c6b4127bR182-R191

File: dm/checker/checker.go (hunk 4 above)
Comment (lance6716) [reply dm/checker/checker.go:226]: I should add an unit test at this level...

Comment (lance6716) [reply dm/checker/checker.go]: https://github.com/pingcap/tiflow/pull/7331/commits/f7a9ceda8304b9d7e4465b1b73e85b17624c80a2

**Review by buchuitoudegou — APPROVED** [review]

**Pull request merged**